src/main/java/org/gwtproject/tools/cldr/LocaleData.java

```diff
@@ -15,1117 +15,1276 @@
+        for (GwtLocale search : locale.getCompleteSearchList()) {
+            String value = getEntry(category, search, key);
+            if (value != null) {
+                return value;
+            }
+        }
+        return null;
     }
-  }
-
-  public Map<String, Map<String, String>> getAllEntries(String localeName) {
-    GwtLocale locale = localeFactory.fromString(localeName);
-    Map<String, Map<String, String>> result = new HashMap<String, Map<String, String>>();
-    for (Map.Entry<MapKey, Map<String, String>> entry : maps.entrySet()) {
-      Map<String, String> map = entry.getValue();
-      if (entry.getKey().getLocale().equals(locale) && !map.isEmpty()) {
-        result.put(entry.getKey().getCategory(), Collections.unmodifiableMap(entry.getValue()));
-      }
+
+    /**
+     * @return all locales that have some data associated with them.
+     */
+    public Set<GwtLocale> getNonEmptyLocales() {
+        Set<GwtLocale> result = new HashSet<GwtLocale>();
+        for (Map.Entry<MapKey, Map<String, String>> entry : maps.entrySet()) {
+            Map<String, String> map = entry.getValue();
+            // Ignore version entries
+            if ("version".equals(entry.getKey().getCategory()) || map.isEmpty()) {
+                continue;
+            }
+            result.add(entry.getKey().getLocale());
+        }
+        return result;
     }
-    return result;
-  }
-
-  /**
-   * @return all locales present in the CLDR data.
-   */
-  public Set<GwtLocale> getAllLocales() {
-    return Collections.unmodifiableSet(allLocales.keySet());
-  }
-
-  /**
-   * Return all entries in a given category and locale.
-   * 
-   * @param category
-   * @param locale
-   * @return map of keys to localized values
-   */
-  public Map<String, String> getEntries(String category, GwtLocale locale) {
-    MapKey mapKey = new MapKey(category, locale);
-    Map<String, String> map = maps.get(mapKey);
-    if (map == null) {
-      return Collections.emptyMap();
+
+    /**
+     * @param category
+     * @return all locales that have some data associated with them in the
+     * specified category.
+     */
+    public Set<GwtLocale> getNonEmptyLocales(String category) {
+        Set<GwtLocale> result = new HashSet<GwtLocale>();
+        for (Map.Entry<MapKey, Map<String, String>> entry : maps.entrySet()) {
+            Map<String, String> map = entry.getValue();
+            if (!category.equals(entry.getKey().category) || map.isEmpty()) {
+                continue;
+            }
+            result.add(entry.getKey().getLocale());
+        }
+        return result;
     }
-    return Collections.unmodifiableMap(map);
-  }
-
-  /**
-   * Returns the keys of all entries in a given cateogry and locale.
-   */
-  List<String> getKeys(String category, GwtLocale locale) {
-    MapKey mapKey = new MapKey(category, locale);
-
-    Map<String, String> map = maps.get(mapKey);
-    if (map == null) {
-      return new ArrayList<String>();
```

Comment:
seems odd that this doesnt return Collections.emptyList(), given that getEntries returns immutables